### ISSUE TITLE
Rework predict() and score() functions to no longer require training data

### ISSUE THREAD

**Issue by rhiever** [rhiever/tpot#92]
`TPOT.predict()` currently requires 3 parameters to produce the predictions: the training and testing features, and the training labels. This breaks the common interface that TPOT shared with sklearn. Let's rework this interface to only require the testing features, and have TPOT store the training features and classes internally on a `fit()` call.

Do the same for `TPOT.score()`.
